test/testcafe/spec/DeviceChallengePollView_spec.js

```diff
@@ -0,0 +1,53 @@
+import { RequestLogger, RequestMock, Selector } from 'testcafe';
+import DeviceChallengePollPageObject from '../framework/page-objects/DeviceChallengePollPageObject';
+import identifyWithDeviceProbingLoopback from '../../../playground/mocks/idp/idx/data/identify-with-device-probing-loopback';
+import loopbackChallengeNotReceived from '../../../playground/mocks/idp/idx/data/identify-with-device-probing-loopback-challenge-not-received';
+
+let failureCount = 0;
+
+const logger = RequestLogger(/introspect|probe|challenge/);
+
+const mock = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(identifyWithDeviceProbingLoopback)
+  .onRequestTo('http://localhost:3000/idp/idx/authenticators/poll')
+  .respond((req, res) => {
+    res.statusCode = '200';
+    if (failureCount === 2) {
+      res.setBody(loopbackChallengeNotReceived);
+    } else {
+      res.setBody(identifyWithDeviceProbingLoopback);
+    }
+  })
+  .onRequestTo('http://localhost:2000/probe')
+  .respond(null, 500, { 'access-control-allow-origin': '*' })
+  .onRequestTo('http://localhost:6511/probe')
+  .respond(null, 500, { 'access-control-allow-origin': '*' })
+  .onRequestTo('http://localhost:6512/probe')
+  .respond(null, 200, { 'access-control-allow-origin': '*' })
```

Comment:
Just curious, how do we know the request with port 6512 onwards will give us a 200 ?

Reply:
I am not sure if I follow your question. Ports will be tried one after another until one responds 200. In this mock, there are ports 2000, 6511, 6512, 6513. Port 2000 and 6512 return 500, port 6512 returns 200. Thus port 6513 won't be used. Port 6512 will continue be used to make a request to /challenge.